### ISSUE TITLE
contentLocation parameter is undefined

### ISSUE THREAD

**Issue by unlight** [mikaturunen/ng-onload#6]
ng-onload version 0.2.1
angular 1.5.6

```
<iframe ng-onload="vm.onFrameLoaded()" ng-src="{{vm.src}}"></iframe>
```

```
        vm.onFrameLoaded = function (contentLocation) {
           // contentLocation empty
            debugger;
            vm.dataType = true;
        };
```

**Comment by unlight** [comment]
> it's undefined when contentWindow cannot be found from the bound element

**Comment by unlight** [comment]
But I see that `contentLocation` object is defined, and on next call it does not pass to callback.
![](https://i.gyazo.com/06116e74afc5f4a302dcb258b621d986.png)

![](https://i.gyazo.com/cf0c4bec20c3976307a9a6ebb7f549b5.png)

`l = Object {contentLocation: Location}`

**Comment by mikaturunen** [comment]
Currently on vacation, I'll look into this once I return so give me a bit of time. If you could, can you post me code that I can use to completely replicate this? Probably the best way to do this is `jsbin`, `jsfiddle` or the sorts.

**Comment by unlight** [comment]
It is copy/paste from readme example:
https://plnkr.co/edit/gHPuPl0X4qyzSzPMe4nw

**Comment by mikaturunen** [comment]
Thanks for the plunker. I did some updates on it against an experimental branch:

https://plnkr.co/edit/xDBMX4OOLb7m3Q4xaSPu?p=preview

If you open up the console, you can see the error I think you're running into with your own application, which is this: `Uncaught SecurityError: Blocked a frame with origin "https://run.plnkr.co" from accessing a frame with origin "https://plnkr.co". Protocols, domains, and ports must match.`. I could be wrong but that's what I'm guessing it is. If you fix the protocols, domains and ports - it should work. It's all related to how http origin and iframes work (if it's this what's causing the issue).

If you git clone the repo and run the following commands in the repo:

```
npm install -g http-server
http-server ./
# open up browser in localhost:8080/test/index.test.html
# open up console and you can see the locations matching properly
```

Example of what it displays as the contentLocation:

```
{
    "hash": "",
    "search": "",
    "pathname": "/test/index.test.html",
    "port": "8080",
    "hostname": "127.0.0.1",
    "host": "127.0.0.1:8080",
    "protocol": "http:",
    "origin": "http://127.0.0.1:8080",
    "href": "http://127.0.0.1:8080/test/index.test.html",
    "ancestorOrigins": {
        "0": "http://127.0.0.1:8080"
    }
}
```

**Comment by unlight** [comment]
https://github.com/mikaturunen/ng-onload/blob/fix-%236/test/index.test.html#L13
This dumps to console `location` var, which is referencing to `window.location`
But the issue was related to callback argument `contentLocation` (iframe.contentWindow.location)
If I change log to `contentLocation` it logs as undefined.

According to readme it can be non-undefined, but how is it possible if in the view template we do not pass any parameter to callback.

**Comment by mikaturunen** [comment]
It was actually working as intended, this is a poorly documented feature on Angular documentation but it takes advantage of the `named function parameters` in Angular. I updated the `README.md` and the index.test.html and now it works as it should :) The missing piece was originating in the README.md, here: https://github.com/mikaturunen/ng-onload/blame/8c793e7188af04a89082019b5f840ad41eb56659/README.md#L58

I have fixed it now it works as it should after the README.md update: https://github.com/mikaturunen/ng-onload/blame/master/README.md#L59

I'm also doing a new release while I'm at it. Did some minor clean ups :) Thanks for the issue report. 

Also note that the Plunker you provided works perfectly "fine", as far as it can work, when you update the index.html: https://plnkr.co/edit/2VkahAutSwI47oiDZWz5?p=preview

So why are you getting an Location object with just `assign` and `replace` in plunker? Because of the earlier mentioned `Uncaught SecurityError: Blocked a frame with origin "https://run.plnkr.co" from accessing a frame with origin "https://plnkr.co". Protocols, domains, and ports must match.`. It will not work in the plunker as the origins are not matching. 

I have also updated the `index.test.html` to show it works now. It would probably do good to notify the ng-onload user that the protocols are mismatching as it currently just assumes the user knows how to play with iframes. 

Hope this helped and thanks for the report again. 

EDIT #: adding reference to the index file that demo's the working functionality: https://github.com/mikaturunen/ng-onload/blob/master/test/index.test.html#L22

**Comment by mikaturunen** [comment]
Alright, all updates in places and the release is available as `0.2.6`. Reopen the issue if the issue resurfaces.
